zerver/worker/queue_processors.py

```diff
@@ -600,24 +597,30 @@ def maybe_send_batched_emails(self) -> None:
             self.timer_event = None
 
             current_time = time.time()
-            for user_profile_id, timestamp in list(self.batch_start_by_recipient.items()):
-                if current_time - timestamp < self.BATCH_DURATION:
+
+            all_user_batches = MissedMessageEmailBatch.objects.all().select_related()
+            for user_batch in all_user_batches:
+                user_profile_id: int = user_batch.user_profile_id
+                events: List[Dict[str, Any]] = user_batch.events
+
+                next_process_time: datetime.datetime = user_batch.next_process_time
+                if current_time - time.mktime(next_process_time.timetuple()) < self.BATCH_DURATION:
```

Comment:
Would it be cleaner to do the `mktime` / `timetuple` stuff on the previous line?

Reply:
I changed the `event_time` to be `datetime.datetime` instead of a `float`, so the conversion is not necessary now.

Reply:
OK, after resolving https://github.com/zulip/zulip/pull/19119#discussion_r664090492, this code doesn't exist now 🙂